xicam/SAXS/data/ensemble.py

```diff
@@ -0,0 +1,85 @@
+from itertools import count
+
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QStandardItem, QStandardItemModel
+
+from xicam.SAXS.widgets.items import CheckableItem
+
+
+class Ensemble:
+    _count = count(1)
+
+    def __init__(self, name=""):
+        super(Ensemble, self).__init__()
+
+        self.catalogs = []
+        self._name = name
+        self._count = next(self._count)
+
+    @property
+    def name(self):
+        if not self._name:
+            self._name = f"Ensemble {self._count}"
+        return self._name
+
+    @name.setter
+    def name(self, name):
+        if not name:
+            return
+        self._name = name
+
+    def append_catalog(self, catalog):
+        self.catalogs.append(catalog)
+
+    def append_catalogs(self, *catalogs):
+        for catalog in catalogs:
+            self.append_catalog(catalog)
+
+
+# class EnsembleItem(QStandardItem):
+#     def __init__(self, *args, **kwargs):
+#         super(EnsembleItem, self).__init__(*args, **kwargs)
+#
+#         self.setCheckable(True)
+#
+#
+
+class EnsembleModel(QStandardItemModel):
+    def __init__(self, *args, **kwargs):
+        super(EnsembleModel, self).__init__(*args, **kwargs)
+
+    def add_ensemble(self, ensemble: Ensemble):
```

Comment:
GUIPlugin should determine how to add catalogs to existing/new ensembles.

Need to pass some kwargs into GUIPlugin.appendCatalog to accomplish this.

1. Open second Catalog in a new ensemble (default)
2. Open second Catalog into existing ensemble 

(Ultimately, move catalogs around).

Context menu on file browsers 
open, open_externally exist already
open_into opens into existing ensemble (manage EnsembleModel top-level items)